Trakt/ScheduledTasks/SyncLibraryTask.cs

```diff
@@ -385,140 +465,218 @@ public async Task ExecuteAsync(IProgress<double> progress, CancellationToken can
-                        {
-                            LogTraktResponseDataContract(traktSyncResponse);
-                        }
+                        var episodesToSend = episodes.Take(100).ToList();
+                        dataContracts = await _traktApi.SendEpisodePlaystateUpdates(
+                            episodesToSend,
+                            traktUser,
+                            seen,
+                            cancellationToken)
+                        .ConfigureAwait(false);
+
+                        episodes.RemoveRange(0, 100);
+                        currentProgress += percentPerRequest;
+                        progress.Report(currentProgress);
+
+                        LogTraktResponseDataContract(dataContracts, TraktItemType.episode);
                     }
-                }
-                catch (ArgumentNullException argNullEx)
-                {
-                    _logger.LogError(argNullEx, "ArgumentNullException handled sending episodes to trakt.tv");
+
+                    dataContracts = (await _traktApi.SendEpisodePlaystateUpdates(
+                        episodes,
+                        traktUser,
+                        seen,
+                        cancellationToken)
+                    .ConfigureAwait(false))
+                    .ToList();
+
+                    currentProgress += percentPerRequest;
+                    progress.Report(currentProgress);
+
+                    LogTraktResponseDataContract(dataContracts, TraktItemType.episode);
                 }
                 catch (Exception e)
                 {
-                    _logger.LogError(e, "Exception handled sending episodes to trakt.tv");
+                    _logger.LogError(e, "Error updating episode play states");
                 }
-
-                progress.Report(100);
             }
         }
 
-        private void LogTraktResponseDataContract(TraktSyncResponse dataContract)
+        private void LogTraktResponseDataContract(IEnumerable<TraktSyncResponse> dataContracts, TraktItemType type)
         {
-            _logger.LogDebug("TraktResponse Added Movies: {Count}", dataContract.Added.Movies);
-            _logger.LogDebug("TraktResponse Added Shows: {Count}", dataContract.Added.Shows);
-            _logger.LogDebug("TraktResponse Added Seasons: {Count}", dataContract.Added.Seasons);
-            _logger.LogDebug("TraktResponse Added Episodes: {Count}", dataContract.Added.Episodes);
-            foreach (var traktMovie in dataContract.NotFound.Movies)
+            if (dataContracts.Any())
             {
-                _logger.LogError("TraktResponse not Found: {@TraktMovie}", traktMovie);
-            }
+                foreach (var dataContract in dataContracts)
+                {
+                    if (type is TraktItemType.movie)
+                    {
+                        if (dataContract.Added != null && dataContract.Added.Movies > 0)
```

Comment:
```suggestion
                        if (dataContract.Added?.Movies > 0)
```

These can all be condensed